guidelines/mwm-workflow-spec.md

```diff
@@ -74,7 +74,58 @@ The task object contains the information:
 |args|object|An object that will be available to the task plugin when executing this task. The expected contents differ based on the task type.|
 |ref|Optional[str]|A reference to a [task template](#task-templates). Values provided by the template are overridden by the task's definition.|
 |pre_evaluators|list[[PreEvaluator](#pre-evaluators)]||
-|output|list[[Output](#output)]|List of possible destinations for the output of tasks in this workflow. Informatics gateways can subscribe to notifications of output to these destinations.|
+|destinations|list[[Output](#output)]|List of possible destinations for the output of tasks in this workflow. Informatics gateways can subscribe to notifications of output to these destinations.|
+|artifacts|[ArtifactMap](#artifacts)|Input & output artifacts of this task.
+
+
+##### Artifacts
+
+
+Tasks can be provided with input artifacts, and they generate output artifacts.
+Inputs can contain artifacts generated by previous tasks or the original workflow input.
+
+The root level Task object contains an optional ArtifactMap object:
+
+###### Artifact Map
+
+| Property | Type | Description |
+|------|------|------|
+|input|list[Artifact]|List of artifacts that are needed by this task.
+|output|list[Artifact]|List of artifacts that are generated by this task.
+
+###### Artifact
+Each Artifact contains at least two elements:
+
+| Property | Type | Required | Description |
+|------|------|------|-----|
+|name|str|Always|The name of this artifact.
+|value|Artifact identifier|Required for inputs, optional for outputs.|The context variable for this artifact (see example). If defined for output, that value will be used instead of a Task-generated one.
+
+As you can see in the example below, input artifacts require a _value_. This is a reference to a previously generated artifact, or to `context.input` - a value for an artifact representing the original input that triggered the workflow.
+
+> ##### Note regarding artifact names:
+> it's not necessary for the input name to be identical to the _name_ specified in the output of a previous task. Output names are used to allow access using the context object. The input name should be what the current task expects to receive.
+
+
+Example:
+```json
+"input": [
+    {
+      "name": "input_dicom",
+      "value": "{{ context.executions.image_type_detector.artifacts.dicom }}"
```

Comment:
Can you please give an example for IG uploaded artifacts? DICOM instances and FHIR resources? How do the values look like?

Reply:
There's an example just above. 
In the case of an IG uploaded Dicom, it'll be `{{ context.input }}`, and it'll be a pointer to the dicom directory.